packages/core/js-sdk/src/store/index.ts

```diff
@@ -960,22 +960,20 @@ export class Store {
       query?: SelectParams,
       headers?: ClientHeaders
     ) => {
-      let paymentCollectionId = (cart as any).payment_collection?.id
-      if (!paymentCollectionId) {
-        const collectionBody = {
-          cart_id: cart.id,
-        }
-        paymentCollectionId = (
-          await this.client.fetch<HttpTypes.StorePaymentCollectionResponse>(
-            `/store/payment-collections`,
-            {
-              method: "POST",
-              headers,
-              body: collectionBody,
-            }
-          )
-        ).payment_collection.id
+      const collectionBody = {
+        cart_id: cart.id,
       }
+      const paymentCollectionId = (
+        await this.client.fetch<HttpTypes.StorePaymentCollectionResponse>(
+          `/store/payment-collections`,
+          {
+            method: "POST",
+            headers,
+            body: collectionBody,
+          }
+        )
+      ).payment_collection.id
```

Comment:
sugg: you can return this function directly and remove the Get from underneath

Reply:
Are they equivalent though? One is a payment collection the other is a payment session. 

Before my changes they were still called together but only when the payment collection id was not defined.